--- ../src-base/minecraft/net/minecraft/client/resources/ResourcePackListEntryDefault.java
+++ ../src-work/minecraft/net/minecraft/client/resources/ResourcePackListEntryDefault.java
@@ -84,7 +84,7 @@
 
     protected String func_148312_b()
     {
-        return "Default";
+        return I18n.func_135052_a("mc.chinaversion.resourcedefalutname", new Object[0]);
     }
 
     protected void func_148313_c()
