--- ../src-base/minecraft/net/minecraft/client/gui/GuiMainMenu.java
+++ ../src-work/minecraft/net/minecraft/client/gui/GuiMainMenu.java
@@ -25,6 +25,7 @@
 import net.minecraft.world.demo.DemoWorldServer;
 import net.minecraft.world.storage.ISaveFormat;
 import net.minecraft.world.storage.WorldInfo;
+import net.minecraftforge.client.ForgeHooksClient;
 import org.apache.commons.io.Charsets;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
@@ -177,7 +178,7 @@
 
         this.field_146292_n.add(new GuiButton(0, this.field_146294_l / 2 - 100, i + 72 + 12, 98, 20, I18n.func_135052_a("menu.options", new Object[0])));
         this.field_146292_n.add(new GuiButton(4, this.field_146294_l / 2 + 2, i + 72 + 12, 98, 20, I18n.func_135052_a("menu.quit", new Object[0])));
-        this.field_146292_n.add(new GuiButtonLanguage(5, this.field_146294_l / 2 - 124, i + 72 + 12));
+        //this.buttonList.add(new GuiButtonLanguage(5, this.width / 2 - 124, i + 72 + 12));
         Object object = this.field_104025_t;
 
         synchronized (this.field_104025_t)
@@ -201,8 +202,8 @@
         fmlModButton.field_146128_h = this.field_146294_l / 2 + 2;
         realmsButton.field_146120_f = 98;
         fmlModButton.field_146120_f = 98;
-        this.field_146292_n.add(realmsButton);
-        this.field_146292_n.add(fmlModButton);
+        //this.buttonList.add(realmsButton);
+        //this.buttonList.add(fmlModButton);
     }
 
     private void func_73972_b(int p_73972_1_, int p_73972_2_)
@@ -478,8 +479,11 @@
         }
         else
         {
-            this.func_73729_b(k + 0, b0 + 0, 0, 0, 155, 44);
-            this.func_73729_b(k + 155, b0 + 0, 0, 45, 155, 44);
+            //this.drawTexturedModalRect(k + 0, b0 + 0, 0, 0, 155, 44);
+            //this.drawTexturedModalRect(k + 155, b0 + 0, 0, 45, 155, 44);
+            Gui.func_152125_a(k + 20, 30, 0, 0, 256, 128, 128, 64, 256, 256);
+            Gui.func_152125_a(k + 120, 30, 0, 128, 256, 128, 128, 64, 256, 256);
+
         }
 
         tessellator.func_78378_d(-1);
@@ -489,7 +493,7 @@
         float f1 = 1.8F - MathHelper.func_76135_e(MathHelper.func_76126_a((float)(Minecraft.func_71386_F() % 1000L) / 1000.0F * (float)Math.PI * 2.0F) * 0.1F);
         f1 = f1 * 100.0F / (float)(this.field_146289_q.func_78256_a(this.field_73975_c) + 32);
         GL11.glScalef(f1, f1, f1);
-        this.func_73732_a(this.field_146289_q, this.field_73975_c, 0, -8, -256);
+        //this.drawCenteredString(this.fontRendererObj, this.splashText, 0, -8, -256);
         GL11.glPopMatrix();
         String s = "Minecraft 1.7.10";
 
@@ -498,16 +502,17 @@
             s = s + " Demo";
         }
 
-        List<String> brandings = Lists.reverse(FMLCommonHandler.instance().getBrandings(true));
-        for (int i = 0; i < brandings.size(); i++)
-        {
-            String brd = brandings.get(i);
-            if (!Strings.isNullOrEmpty(brd))
-            {
-                this.func_73731_b(this.field_146289_q, brd, 2, this.field_146295_m - ( 10 + i * (this.field_146289_q.field_78288_b + 1)), 16777215);
-            }
-        }
-        String s1 = "Copyright Mojang AB. Do not distribute!";
+        //List<String> brandings = Lists.reverse(FMLCommonHandler.instance().getBrandings(true));
+        //for (int i = 0; i < brandings.size(); i++)
+        //{
+        //    String brd = brandings.get(i);
+        //    if (!Strings.isNullOrEmpty(brd))
+        //    {
+        //        this.drawString(this.fontRendererObj, brd, 2, this.height - ( 10 + i * (this.fontRendererObj.FONT_HEIGHT + 1)), 16777215);
+        //    }
+        //}
+        ForgeHooksClient.renderMainMenu(this, field_146289_q, field_146294_l, field_146295_m);
+        String s1 = I18n.func_135052_a("mc.chinaversion.copyright", new Object[0]);
         this.func_73731_b(this.field_146289_q, s1, this.field_146294_l - this.field_146289_q.func_78256_a(s1) - 2, this.field_146295_m - 10, -1);
 
         if (this.field_92025_p != null && this.field_92025_p.length() > 0)
